pkg/gadgets/bpftrace/tracer.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build !withoutebpf
+
+package tracer
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"os/exec"
+
+	gadgetcontext "github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-context"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/bpftrace/types"
+)
+
+type Config struct {
+	Program string
+}
+
+type Tracer struct {
+	eventCallback func(ev *types.Event)
+	config        Config
+}
+
+func (t *Tracer) Run(gadgetCtx gadgets.GadgetContext) error {
+	flavour := os.Getenv("GADGET_IMAGE_FLAVOUR")
+	if flavour != "default" {
+		return fmt.Errorf("bpftrace is not supported on the %q flavour of the container image. Only default is supported for now",
+			flavour)
+	}
+
+	params := gadgetCtx.GadgetParams()
+	t.config.Program = params.Get(ParamProgram).AsString()
+	log := gadgetCtx.Logger()
+
+	cmd := exec.Command("bpftrace", "-e", t.config.Program)
```

Comment:
What about using `CommandContext`, starting the go routines for the stdout and stderr, and then just calling `cmd.Run()`. Having used `CommandContext`, the `Run()` will automatically start the command, wait for its completion or kill the process when the context is done. WDYT?

Reply:
Before golang 1.20, the process is only killed when then context is done and it hasn't terminated, in this case we don't want to kill the process but interrupt it (otherwise it won't produce the output). In golang 1.20, it's possible to have a custom cancel function to interrupt the process. I added a note there to study that possibility once we supoprt go 1.20.